### PR TITLE
Remove replace of go-openapi/spec from go.mod

### LINKED ISSUE #1439: Remove `go-openapi/spec` replace directive in go.mod
We should be able to remove `go-openapi/spec` replace directive since k8s 0.21 newer `kustomize` finally.

_Originally posted by @dsimansk in https://github.com/knative/client/pull/1434#discussion_r698408433_

/kind cleanup

### PR DESCRIPTION
**Pull request by dsimansk**
## Changes

* Update `k8s.io/cli-runtime` to `v0.21.4` to rest of k8s.io versions
* Remove unnecessary `go-openapi/spec` replace

## Reference

Fixes #1439

### CODE REVIEW

File: docs/cmd/kn_domain_describe.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ kn domain describe NAME
   -h, --help                          help for describe
   -n, --namespace string              Specify the namespace to operate in.
   -o, --output string                 Output format. One of: json|yaml|name|go-template|go-template-file|template|templatefile|jsonpath|jsonpath-as-json|jsonpath-file|url.
+      --show-managed-fields           If true, keep the managedFields when printing objects in JSON or YAML format.
```
Comment (rhuss) [inline docs/cmd/kn_domain_describe.md:24]: Does this work ? So are managed fields hidden by default now ?

Comment (dsimansk) [reply docs/cmd/kn_domain_describe.md:24]: Yes, it does. The change is part of updated `k8s.io/cli-runtime v0.21.4`. The latest `kubectl` hides managed fields as well.

E.g.:
```
➜  client git:(pr/cleanup-replace) kn service describe hello -o yaml --show-managed-fields
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  annotations:
    serving.knative.dev/creator: kubernetes-admin
    serving.knative.dev/lastModifier: kubernetes-admin
  creationTimestamp: "2021-09-02T10:53:19Z"
  generation: 1
  managedFields:
  - apiVersion: serving.knative.dev/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:spec:
        .: {}
        f:template:
          .: {}
          f:metadata:
            .: {}
            f:annotations:
              .: {}
              f:client.knative.dev/updateTimestamp: {}
              f:client.knative.dev/user-image: {}
            f:creationTimestamp: {}
          f:spec:
            .: {}
            f:containers: {}

```
vs.
```
➜  client git:(pr/cleanup-replace) kn service describe hello -o yaml                      
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  annotations:
    serving.knative.dev/creator: kubernetes-admin
    serving.knative.dev/lastModifier: kubernetes-admin
  creationTimestamp: "2021-09-02T10:53:19Z"
  generation: 1
  name: hello
  namespace: default
  resourceVersion: "222953"
  uid: 95036efa-c9f9-4065-b1c0-f7794d0b1ad0
```

**Review by maximilien — COMMENTED** [review]
Is the title of this PR a misnomer? Why would replacing “go-openapi/spec” cause changes to any command? Maybe I am missing something…

**Comment by rhuss** [comment]
> Is the title of this PR a misnomer? Why would replacing “go-openapi/spec” cause changes to any command? Maybe I am missing something…

It's that we pinned the version for this dependency with a `replace` directive in the go.mod

```
replace github.com/go-openapi/spec => github.com/go-openapi/spec v0.19.3
```

This was introduced to avoid a bug from cli-runtime, that is fixed in a newer version.


The PR looks good to me

/approve
/lgtm

**Comment by dsimansk** [comment]
> Is the title of this PR a misnomer? Why would replacing “go-openapi/spec” cause changes to any command? Maybe I am missing something…

Well, the title isn't the most descriptive. To explain the changes. The rest of Knative is bumping to k8s.io deps `0.21.4`, therefore we're able to use newer k8s.io/cli-runtime. That version has sorted out transitive dependency `go-openapi/spec` that had to keep locked with `replace`. The new flag `--show-managed-fields` is a new feature of `cli-runtime`, hence the changed docs etc.

**Pull request merged**